### PR TITLE
Pyodide wheel

### PR DESCRIPTION
**Pull request by texodus**
Adds a pyodide-built wasm32 wheel of `perspective-python`.

### CODE REVIEW

File: cmake/arrow/CMakeLists.txt [hunk 1]
```diff
@@ -277,11 +277,7 @@ target_compile_definitions(arrow PUBLIC ARROW_STATIC)
 
 # will need built boost filesystem and system .lib to work, even though
 # perspective itself does not use those dependencies
-target_link_libraries(arrow
-    ${double-conversion_LIBRARIES}
-    ${Boost_FILESYSTEM_LIBRARY}
-    ${Boost_SYSTEM_LIBRARY}
-    ${ARROW_TEST_LINK_TOOLCHAIN})
+target_link_libraries(arrow)
```
Comment (timkpaine) [inline cmake/arrow/CMakeLists.txt:280]: This doesn't do anything anymore, are we sure `double-conversion` isn't required?

File: cpp/perspective/CMakeLists.txt [hunk 2]
```diff
@@ -1,8 +1,7 @@
-cmake_minimum_required(VERSION 3.7.2)
+cmake_minimum_required(VERSION 3.18.2)
 project(psp)
 include(CheckCCompilerFlag)
 
-set(CMAKE_BUILD_TYPE "Release")
```
Comment (timkpaine) [inline cpp/perspective/CMakeLists.txt:5]: This can stay

File: python/perspective/setup.py [hunk 3]
```diff
@@ -139,13 +139,8 @@ def build_extension_cmake(self, ext):
             "-DPSP_WASM_BUILD=0",
             "-DPSP_PYTHON_BUILD=1",
             "-DPSP_PYTHON_VERSION={}".format(PYTHON_VERSION),
-            "-DPython_ADDITIONAL_VERSIONS={}".format(PYTHON_VERSION),
```
Comment (timkpaine) [inline python/perspective/setup.py:142]: are we sure that these are all not necessary? Did you test under a virtualenv

https://github.com/finos/perspective/commit/61557c6a28a0c70e00cd4ee9160ef22e65c2ad99

File: python/perspective/setup.py [hunk 4]
```diff
@@ -139,13 +139,8 @@ def build_extension_cmake(self, ext):
...
-            "-DPython_FIND_VERSION={}".format(PYTHON_VERSION),
             "-DPython_EXECUTABLE={}".format(sys.executable).replace("\\", "/"),
-            "-DPYTHON_LIBRARY={}".format(sysconfig.get_config_var("LIBDIR")).replace("\\", "/"),
-            "-DPYTHON_INCLUDE_DIR={}".format(sysconfig.get_config_var("INCLUDEPY")).replace("\\", "/"),
-            "-DPython_ROOT_DIR={}".format(sys.prefix).replace("\\", "/"),
-            "-DPython_ROOT={}".format(sys.prefix).replace("\\", "/"),
+            "-DPYTHON_INCLUDE_DIR={}".format(sysconfig.get_path("include")).replace("\\", "/"),
```
Comment (timkpaine) [inline python/perspective/setup.py:143]: this one in particular needs to be tested to not regress https://github.com/finos/perspective/commit/61557c6a28a0c70e00cd4ee9160ef22e65c2ad99

**Comment by timkpaine** [comment]
this needs a full build kicked off

**Comment by texodus** [comment]
[full build](https://github.com/finos/perspective/actions/runs/5166313137/jobs/9306461272)

**Pull request merged**